main.py

```diff
@@ -0,0 +1,124 @@
+import requests
+import re
+from lxml import html
+from lxml.cssselect import CSSSelector
+import csv
+
+urlinit = 'http://www.schooldays.ie'
+
+def regionurl():
+
+#Collect the content of each schooldays.ie urls for region Dublin-1, Dublin-2 ....Dublin-18 and calls schollurl() function,
+#Create the CSV file with the header
+
+    header = ['School Name', 'Address', 'Postal District', 'Year', 'Total Sitting Leaving', #csv reader
+          'Total Main Colleges']
+    with open('./schoolfile', 'a') as myfile: #write the csv reader
+        wr = csv.writer(myfile, quoting=csv.QUOTE_ALL)
+        wr.writerow(header)
+
+    dublinregions = list(range(1,19))
+    otherdublinregions = ['6w', '20', '22', '24', 'north-county-dublin', 'south-county-dublin', 'west-county-dublin']
+    allregions = dublinregions+otherdublinregions # all the dublin postal codes /regions
+
+    for i in allregions:
+
+        urlregioninitial = urlinit+'/secondary-schools-in-ireland/Dublin-' + str(i)
```

Comment:
This is a personal opinion. I prefer to concatenate strings using .format.. I think it looks more neat:

```
urlregioninitial = '{}/secondary-schools-in-ireland/Dublin-/{}'.format(urlinit, i)
```
and even if you go in the former option... you should have spaces between operators.

Reply:
ok